Date  Tue, 11 May 2004 04:57:00 +0100 (IST)
From  Dave Airlie <>
Subject  [patch] serial fifo size is ignored

When using register_serial the xmit_fifo_size parameter is accepted
by the 8250 driver, and copied to the uart_port fifosize parameter,
however autoconfigure then comes along and overrides this from
the dfl_xmit_fifo_size, this patch checks if fifosize is 0
and if it is updates from the default, otherwise it accepts the value,

--- linux-2.6.9-rc2/drivers/serial/8250.c.orig	2004-09-13 07:31:29.000000000 +0200
+++ linux-2.6.9-rc2/drivers/serial/8250.c	2004-09-14 10:14:11.653931304 +0200
@@ -733,7 +733,8 @@
 #endif
 	serial_outp(up, UART_LCR, save_lcr);
 
-	up->port.fifosize = uart_config[up->port.type].fifo_size;
+	if (up->port.fifosize == 0)
+		up->port.fifosize = uart_config[up->port.type].fifo_size;
 	up->capabilities = uart_config[up->port.type].flags;
 	up->tx_loadsz = uart_config[up->port.type].tx_loadsz;
 
